app/models/robot_directory.rb

```diff
@@ -1,59 +1,101 @@
+require 'faker'
+
 class RobotDirectory
   attr_reader :directory
 
-  def initialize(directory)
-    @directory = directory
+  def initialize(database)
+    @directory = database
   end
 
   def create(robot)
-    directory.transaction do
-      directory['robots'] ||= []
-      directory['total'] ||= 0
-      directory['total'] += 1
-      directory['robots'] << { 'serial_number' => directory['total'] * rand(100),
-                               'name' => robot[:name],
-                               'city' => robot[:city],
-                               'state' => robot[:state],
-                               'avatar' => robot[:avatar],
-                               'birthdate' => robot[:birth_date],
-                               'date_hired' => robot[:date_hired],
-                               'department' => robot[:department] }
-    end
+    directory.execute("INSERT INTO robots
+      (name, city, state, avatar, birthdate, date_hired, department)
+      VALUES
+      (?, ?, ?, ?, ?, ?, ?);", robot[:name], robot[:city], robot[:state],
+      'robohash', robot[:birthdate], robot[:date_hired], robot[:department])
+
+    serial = directory.execute("SELECT serial_number
```

Comment:
what happens if two robot's have the same name? will this still work? 

i'd suggest maybe grabbing the last inserted row, then finding the serial number from that record

also - there's a LOT going on in this method. i encourage you to extract some of these pieces out and encapsulate them in other methods. maybe a `find_last_inserted_robot` and a `update_robot_avatar` method?